playground/src/pages/CodingEnvironment.tsx

```diff
@@ -15,6 +16,96 @@ async function getExample(example: string) {
 
 function CodingEnvironment() {
 
+  /**
+  * Helper function to check if input contains only numbers.
+  */ 
+   function checkInputField(jsonField: JSON, key: string) {
+
+    const jsonInput = jsonField[key as keyof typeof jsonField];
+
+    if (!Array.isArray(jsonInput)) {
+
+      const errorMessage = `If you use ${key},
+      it must be an array of numbers.`;
+
+      toast.error(errorMessage);
+
+      setOutput(errorMessage);
+
+      return false;
+
+    }
+
+    if (Object.values(jsonInput).length === 0 || Object.values(jsonInput).some(isNaN)) {
+
+      const errorMessage = `If you use ${key}, 
+      it must contain at least one number, 
+      and it can only contain numbers.`;
+
+      toast.error(errorMessage);
+
+      setOutput(errorMessage);
+
+      return false;
+    }
+
+    return true;
+
+    }
+
+
+  /**
+  * We check the inputs and return true or false. We allow: 
+  * - an empty input, and
+  * - a valid JSON object containing stack_init or advice_tape (or both) 
+  * if the values are numbers
+  */ 
+  function checkInputs(jsonString: string) {
+
+    if (jsonString === "") {
+      return true;
+    }
+
+    let jsonInput!: JSON;
+
+    try {
+
+      jsonInput = JSON.parse(jsonString);
+
+    }
+    catch (e) {
+
+      const errorMessage = `Miden VM Inputs need to be 
```

Comment:
This is working, but the error message is not great.

Could you please add the contents of `e`? It is a [`SyntaxError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError), `e.message` should be enough.

Reply:
Isn't the errorMessage defined here the better approach? The SyntaxError is caused when the input is not a valid JSON, right?

Reply:
Got it, all good now.

I will add tests for all cases in #28 - which will be the next PR